pytorch_lightning/core/lightning.py

```diff
@@ -1548,8 +1548,9 @@ def _load_model_state(cls, checkpoint: Dict[str, Any], *args, **kwargs) -> 'Ligh
                 )
 
         # load the state_dict on the model automatically
-        model_args = [hparams] if hparams else []
-        model = cls(*model_args, *args, **kwargs)
+        if hparams:
+            kwargs.update(hparams=hparams)
+        model = cls(*args, **kwargs)
```

Comment:
I see what you are trying to do but it fails now when hparams actually IS a positional argument. See this simple example:
```python
def foo(hparams, x, y=1):
    print(hparams, x, y)

foo(1, y=1, hparams={})  # results in TypeError
```

Reply:
I think there is currently no way around it if we want to auto save and load the hparams. It must be called hparams and it must be a positional argument (or no hparams argument at all).

Reply:
The example you provided raise error because hparams has been set twice, but throughout this package, I didn't find anywhere else that call the constructor of the module. I think there's isn't any case where user call the instructor while the package also provide hparams.

Reply:
Ok, I see my error, passing the args with *args **kwargs is ok because kwargs can contain the positional ones too. 
My comment was because I thought this was the reason the tests are failing. Must be something else then. I will checkout the branch and look at the test.